TypeScript.py

```diff
@@ -1501,6 +1523,81 @@ def run(self, text):
             formatRange(text, view, rblineStart, ralineEnd)
 
 
+class NavToCommand(sublime_plugin.WindowCommand):
+    # indicate the nav to panel is started or not
+    started = False
+    # indicate if the insert_text command has finished pasting text into the textbox.
+    # during which time the on_modified callback shouldn't do angthing
+    ini_finished = False
+    reset_already = False
+    search_text = ""
+
+    @classmethod
+    def reset(cls):
+        cls.started = False
+        cls.ini_finished = False
+
+    def is_enabled(self):
+        return is_typescript(self.window.active_view())
+
+    def run(self, search_text = ""):
+        logger.log.debug ("start running nvato wiht text: %s" % search_text)
+
+        NavToCommand.reset_already = False
+        
+        self.window.run_command("hide_overlay")
+
+        if NavToCommand.reset_already:
+            NavToCommand.reset_already = False
+        else:
+            logger.log.debug ("reset in run")
+            NavToCommand.reset()
+            NavToCommand.reset_already = True
+
+        NavToCommand.search_text = search_text        
+        NavToCommand.started = True
+        actual_search_text = "a" if search_text == "" else search_text
+        self.items = cli.service.navTo(actual_search_text, self.window.active_view().file_name())
+        self.window.show_quick_panel(self.format_navto_res(self.items), self.on_done)  
+        logger.log.debug ("end running nvato wiht text: %s" % search_text)
+        
+    def on_done(self, index):
+        logger.log.debug ("enter on_done. search_text: %s, started: %s, init_finished: %s" % 
+               (NavToCommand.search_text, NavToCommand.started, NavToCommand.ini_finished))
+        
+        if NavToCommand.reset_already:
+            NavToCommand.reset_already = False
+        else:
+            logger.log.debug ("reset in on_done")
+            NavToCommand.reset()
+            NavToCommand.reset_already = True
+
+        if index >= 0:
+            item = self.items[index]
+            line, offset = item['start']['line'], item['start']['offset']
+            self.window.open_file(item['file'] + ":%s:%s" % (line, offset), sublime.ENCODED_POSITION)
```

Comment:
OR preferably, make a command that formats this for you and use it everywhere you call this